File: easy/1. Two Number Sum/two_number_sum.go
package main

func TwoNumberSum(array []int, target int) []int {
	m := make(map[int]int)
	n_len := len(array)

	for i := 0; i < n_len; i++ {
		num := array[i]
		diff := target - num
		if _, ok := m[diff]; ok {
			return []int{diff, num}
		}
		m[num] = i
	}

	return []int{}
}

File: easy/1. Two Number Sum/two_number_sum.cpp
#include <iostream>
#include <map>
#include <vector>

std::vector<int> twoNumberSum(std::vector<int> array, int targetSum) {
  std::map<int, int> dict;
  for (int i = 0; i < array.size(); i++) {
    int num = array[i];
    int diff = targetSum - num;
    if (dict.find(diff) != dict.end()) {
      return std::vector<int> {diff,num};
    }
    dict[num] = i;
  }
  return {};
}

int main() { return 0; }
